### PR TITLE
Don't sort an empty file list

### LINKED ISSUE #1: sort hangs if sortFiles gets an empty list
https://github.com/kartotherian/tilerator-jobprocessor/blob/master/lib/fileParser.js#L116 will hang on stdin if the list if empty.

Related-To: https://phabricator.wikimedia.org/T145534#2987786

### PR DESCRIPTION
**Pull request by MaxSem**
sort just waits for stdin forever

Closes #1

Bug: T145534

### CODE REVIEW

File: lib/fileParser.js [hunk 1]
```diff
@@ -268,13 +271,17 @@ function fileParser(filepath, options, addJobCallback) {
...
+
                 return sortFile(tempFiles2, cleanupList);
             }).then(function (sortedFile) {
                 cleanupList.push(sortedFile);
                 return addJobsFromSortedFile(sortedFile, options, addJobCallback);
             }).then(function (result) {
                 core.log('info', 'Finished job creation... no more unemployment... at least for a bit');
                 return result;
+            }).catch(function() {
+                core.log('info', 'All tiles filtered out, nothing to process');
```
Comment (nyurik) [inline lib/fileParser.js]: I don't think catching and ignoring all errors is good. Its better to explicitly handle it by returning `false` or `null`, and handling it.

**Review by nyurik — CHANGES_REQUESTED** [review]

**Comment by MaxSem** [comment]
Updated.

**Comment by nyurik** [comment]
I'm a bit confused - you still throw an exception if the sort list is empty, but you handle it later too?

**Comment by MaxSem** [comment]
The check in `sortFile()` is needed to prevent any reoccurences in the future - this function can still hang if called with empty list.

**Comment by MaxSem** [comment]
Resolved by @nyurik in 4186abc5b2752c9ea5fccbb1dcb638502a572959.

**Pull request closed without merging**